recipes/repository.rb

```diff
@@ -29,7 +29,7 @@
   # Trust new APT key
   execute 'apt-key import key 382E94DE' do
     command 'apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 A2923DFF56EDA6E76E55E492D3A80E30382E94DE'
-    not_if 'apt-key list | grep 382E94DE'
+    not_if 'apt-key list | grep "382E 94DE"'
```

Comment:
I think the output of `apt-key list` isn't consistent between versions of `apt`. For instance, on an ubuntu 14.04 VM with `apt` version `1.0.1ubuntu2`, the output is:

```
/etc/apt/trusted.gpg
--------------------

pub   4096R/382E94DE 2016-06-29 [expires: 2022-06-28]
uid                  Datadog, Inc <package@datadoghq.com>
sub   4096R/F432F6E0 2016-06-29 [expires: 2022-06-28]
sub   4096R/8387EEAF 2016-06-29 [expires: 2022-06-28]
sub   4096R/BFF6291E 2016-07-11 [expires: 2022-07-10]
```

So we may want to rely on the output of `apt-key fingerprint` instead since its output might be more stable across `apt` versions.